Add segmentations from William Sirois

mguaypaq opened the pull request:
These are the files provided by William Sirois at the end of his GBM3100 internship. Todo:
- Figure out which `derivatives/` subfolder they should go into.
- Rename the files according to BIDS.
- Add JSON sidecars.

While uploading the files, I noticed that most of them are new, but a few of them are exact duplicates of already existing files in the repository.

mguaypaq committed:
Add files from William Sirois

sandrinebedard commented:
I added to the branch the spinal cord canal seg under derivatives/labels, I am not sure that it worked, @mguaypaq can you check?

sandrinebedard committed:
add canal seg

mguaypaq commented:
It looks like this is what happened:
- 1 of the 188 images from William Sirois (`sub-barcelona02_011`) got replaced by a new image from Sandrine (`sub-barcelona02_space-other_T2w_label-canal_seg.nii.gz`).
- 4 of the 188 images from William Sirois were exact copies of existing `canal_seg` files under `derivatives/labels/`.
- 6 of the 188 images from William Sirois replaced existing `canal_seg` files under `derivatives/labels/`.
- The remaining 177 of 188 images from William Sirois are new `canal_seg` files under `derivatives/labels/`.

But:
- There are also 80 new `sub-*_T2w_label-discs_dlabel.nii.gz` images from Sandrine under `derivatives/labels/`.

@sandrinebedard is this what you wanted?

sandrinebedard commented:
> But:
> 
> * There are also 80 new `sub-*_T2w_label-discs_dlabel.nii.gz` images from Sandrine under `derivatives/labels/`.
> 
> @sandrinebedard is this what you wanted?

Nope hahaha, just realized a script I was running was modifying the files... sorry, we should remove those changes

mguaypaq commented:
Ok, I can remove the extra modifications. Do we also need to create and/or update any JSON sidecars with the provenance of the data?

sandrinebedard commented:
Abel just created them yesterday, I will push them today!

sandrinebedard committed:
add json sidecars to canal seg

mguaypaq commented:
I don't think these are the right JSON sidecars; they describe the 267 images mentioned at the end of https://github.com/ivadomed/model-canal-seg/issues/7, but not the 188 images mentioned at the start (which are the ones in this PR). We probably need at least 4 different JSON files:
- The 4 images that were already in this dataset before the PR have existing JSON sidecars, so we should use them.
- The 1 image that was modified in https://github.com/spine-generic/data-multi-subject/pull/172/commits/5c22e3d83da3a9d5cca4cf83e77a511b273a2a4d (instead of being just moved) needs an explanation of what was done to modify it.
- The 6 images that _replace_ existing canal segs, we should check whether they are manual corrections, or the result of an earlier model, or what.
- The 177 other images, we should figure out how they were produced.

That's if we want the 188 images used as ground truth for the training. Alternatively, if we want the 267 images produced by the final model, we could upload them instead? Or, if we're not sure of the origin, we should say that in the JSON files.

mguaypaq committed:
Revert accidental modifications to `*_dlabel.nii.gz` files

mguaypaq committed:
Revert accidental modifications of label provenance

The corresponding image files are unmodified, so their provenance shouldn't be overwritten.

mguaypaq committed:
Note that one label file was modified by Sandrine

mguaypaq committed:
Remove unverified provenance information for labels

It's unclear which model was actually used to generate these images.

mguaypaq committed:
Remove temporary files

Now that these files have been integrated under derivatives/labels/, we don't
need to keep the original non-BIDS names.

mguaypaq commented:
Ok, here's what I did:
- I reverted the accidental modifications to 80 `*_dlabel.nii.gz` files.
- I reverted the JSON changes for the 4 images that were already present (with no modifications) on `master`. Even though these label files were included in William Sirois's zip file, clearly he didn't create them.
- I wrote the following JSON file for `sub-barcelona02_space-other_T2w_label-canal_seg`, since it was modified by Sandrine in this PR compared to William Sirois's zip file:
  ```json
  {
      "SpatialReference": {
          "Resampling": "0.8x0.8x0.8",
          "Reorientation": "RPI"
      },
      "GeneratedBy": [
          {
              "Author": "William Sirois"
          },
          {
              "Author": "Sandrine Bedard",
              "Date": "2024-08-12 00:00:00"
          }
      ]
  }
  ```
- I wrote the following minimal JSON file for the other 183 label files, since it's unclear what model and/or what other commands were actually run to generate them:
  ```json
  {
      "SpatialReference": {
          "Resampling": "0.8x0.8x0.8",
          "Reorientation": "RPI"
      },
      "GeneratedBy": [
          {
              "Author": "William Sirois",
              "Date": "2024-03-17 00:00:00"
          }
      ]
  }
  ```
- I deleted the temporary directory `derivatives/GT_canalseg_WS_GBM3100/` which initially contained the unzipped files.

@sandrinebedard Does this look ok to merge?
